test/test_check_branch.py

```diff
@@ -70,3 +70,18 @@ def expect_exception(branch, deployment, message):
                              'prod',
                              "Branch 'feature/foo' cannot be deployed to 'prod', "
                              "only one of {'sandbox'} or personal deployments.")
+
+        supported_vars = ['CI_COMMIT_REF_NAME', 'GITHUB_BASE_REF', 'GITHUB_HEAD_REF']
+
+        for active_env_var in supported_vars:
+            if active_env_var == 'GITHUB_HEAD_REF':
+                base_branch = ['foo/bar']
+            else:
+                base_branch = ['develop', 'prod']
+            for branch in base_branch:
+                with self.subTest(branch_name=branch, active_env_variable=active_env_var):
+                    with patch.dict(os.environ, {active_env_var: branch}):
+                        for env_var in supported_vars:
+                            if env_var != active_env_var:
+                                os.environ.pop(env_var, None)
```

Comment:
I don't think it is safe to permanently modify `os.environ` this way. That's why we use `patch` in the first place. Please raise in PL.

Reply:
Modifications made within the scope of the patch are _not_ permanent:

```
>>> import os
>>> from unittest.mock import patch
>>> os.environ['project_root']
'/home/nadove/PycharmProjects/azul-sc'
>>> with patch.dict(os.environ):
...     os.environ.pop('project_root')
...     assert 'project_root' not in os.environ
    
>>> os.environ['project_root']
'/home/nadove/PycharmProjects/azul-sc'
```